tests/e2e/macvtap_test.go

```diff
@@ -0,0 +1,301 @@
+							}
+						},
+						"spec":{
+							"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"%s\", \"type\": \"macvtap\"}"
+						}
+					}
+				`
+
+			BeforeEach(func() {
+				clientset.RESTClient().
+					Post().
+					RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+					Body([]byte(fmt.Sprintf(nad, macvtapIfaceName, namespace, getMacvtapResourceName(lowerDevice), macvtapIfaceName))).
+					Do()
+			})
+
+			AfterEach(func() {
+				clientset.RESTClient().
+					Delete().
+					RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+					Do()
+			})
+
+			Context("WHEN a pod requests aforementioned macvtap resource", func() {
+				podName := "megapod"
+				containerName := "tinycontainer"
+
+				container := v1.Container{
+					Name:    containerName,
+					Image:   "alpine",
+					Command: []string{"/bin/sh", "-c", "sleep 999999"},
+					Resources: v1.ResourceRequirements{
+						Limits: getMacvtapResourceRequest(lowerDevice, 1),
+					},
+				}
+
+				pod1 := &v1.Pod{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Pod",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        podName,
+						Namespace:   namespace,
+						Annotations: getMacvtapNetworkAnnotations(macvtapIfaceName),
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{container},
+					},
+				}
+
+				BeforeEach(func() {
+					_, err := clientset.CoreV1().Pods(namespace).Create(pod1)
+					Expect(err).NotTo(HaveOccurred())
+
+					// wait for pod to be ready
+					waitForPodReadiness(podName, namespace, Timeout)
+				})
+
+				AfterEach(func() {
+					err := clientset.CoreV1().Pods(namespace).Delete(podName, nil)
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("THEN the pod successfully gets a second interface, of macvtap type, with configured MAC address", func() {
+					pod, err := clientset.CoreV1().Pods(namespace).Get(podName, metav1.GetOptions{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "Pod",
+							APIVersion: "v1",
+						},
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					// assert resources have been allocated
+					Expect(pod.Annotations).To(HaveKey(NetworkResource))
+					Expect(pod.Spec.Containers).To(HaveLen(1))
+
+					theContainer := pod.Spec.Containers[0]
+					Expect(theContainer.Resources.Limits).To(Equal(getMacvtapResourceRequest(lowerDevice, 1)))
+
+					// assert MAC address is found on the second interface
+					podNetworks := pod.Annotations[NetworkStatus]
+
+					networks, err := parseNetwork(podNetworks)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(networks).To(HaveLen(2))
+
+					// macvtap iface is created as a secondary iface
+					macvtapNetwork := networks[1]
+					Expect(macvtapNetwork.Name).To(Equal(macvtapIfaceName))
+					Expect(macvtapNetwork.Interface).To(Equal("net1"))
+					Expect(macvtapNetwork.Mac).NotTo(BeEmpty())
+				})
+			})
+
+			Context("WHEN a VMI requests aforementioned macvtap resource", func() {
```

Comment:
If you want, you can skip the whole context `PContext`

Reply:
Done.